pkg/cli/values/options.go

```diff
@@ -97,6 +95,28 @@ func (opts *Options) MergeValues(p getter.Providers) (map[string]interface{}, er
 	return base, nil
 }
 
+// mergeYaml merges data into an existing Go map. data must be a YAML format.
+// If the file contains multiple documents, each document will be merged separately.
+func mergeYaml(base map[string]interface{}, data []byte) (map[string]interface{}, error) {
```

Comment:
Helm already has a well defined process for merging yaml files. For example, maps are merged, arrays are replaced, `null` values cause deletion. See: https://github.com/helm/helm/blob/main/pkg/chartutil/coalesce.go I think it would be best to follow existing behavior, and effectively treat a multiple-doc yaml values file as multiple values files.

Also see this (upcoming, presumably) PR, which looks to modify this process some:
https://github.com/helm/helm/pull/11440/files

Reply:
Or, at least, I think the business logic of how to handle multiple yaml documents doesn't belong in options parsing code (given the above business logic around values handling)

Reply:
@gjenkins8 the existing behavior is not changed. The `mergeYaml` function will use the current `mergeMaps` function. The relevant function `mergeMaps` which holds the logic has been moved from line 59 to line 114.

Reply:
ugh, oops. You're right. I missed that the existing logic ie. passing multiple files uses `mergeMaps` already.